File: scripts/node/parsers/applicationParser.js
'use strict';

const fs   = require('fs');
const path = require('path');

function extractTag(xml, tag) {
  const m = xml.match(new RegExp(`<${tag}[^>]*>([\\s\\S]*?)<\\/${tag}>`));
  return m ? m[1].trim() : null;
}

/**
 * Reads Custom App (Lightning App) metadata from SFDX source format.
 * @param {string} typeDir - path to the applications/ folder
 * @returns {{ apps: Array }}
 */
async function parseApplications(typeDir) {
  const apps = [];
  if (!fs.existsSync(typeDir)) return { apps };

  const files = fs.readdirSync(typeDir).filter(f => f.endsWith('.app-meta.xml'));

  for (const file of files) {
    const xml = fs.readFileSync(path.join(typeDir, file), 'utf8');
    apps.push({
      apiName:     file.replace('.app-meta.xml', ''),
      label:       extractTag(xml, 'label'),
      description: extractTag(xml, 'description'),
      navType:     extractTag(xml, 'navType') || 'Standard',
    });
  }

  return { apps };
}

module.exports = { parseApplications };

File: scripts/python/parsers/application_parser.py
"""Parser for Custom App (Lightning App) metadata (.app-meta.xml)."""
import re
from pathlib import Path


def _extract_tag(xml: str, tag: str) -> str | None:
    m = re.search(rf'<{tag}[^>]*>([\s\S]*?)</{tag}>', xml)
    return m.group(1).strip() if m else None


def parse_applications(type_dir: str) -> dict:
    """
    @param type_dir: path to the applications/ folder
    @returns: { apps: list }
    """
    apps = []
    p = Path(type_dir)
    if not p.exists():
        return {'apps': apps}

    for f in sorted(p.glob('*.app-meta.xml')):
        xml = f.read_text(encoding='utf-8')
        apps.append({
            'apiName':     f.name.replace('.app-meta.xml', ''),
            'label':       _extract_tag(xml, 'label'),
            'description': _extract_tag(xml, 'description'),
            'navType':     _extract_tag(xml, 'navType') or 'Standard',
        })

    return {'apps': apps}
